### PR TITLE
Image conversion lib & test

### PR DESCRIPTION
**Pull request by maxkadel**
* includes libvips-tools & imagemagick & libtiff-tools

### CODE REVIEW

File: lib/nga.sh [hunk 1]
```diff
@@ -0,0 +1,134 @@
+#!/bin/bash
```
Comment (maxkadel) [inline lib/nga.sh]: More descriptive file name? It's a bit cryptic.

File: lib/nga.sh [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    if [ -z "${savefiles}" ]  && [ -d $tmpprefix ]; then rm -f $outprefix*; fi
+}
+
+trap 'handle_error ${LINENO} $?' ERR
+
+# cause the script to fail if any commands exist with non-zero status
+set -e
+
+if [[ -z "$1" || -z "$2" || -z "$3" ]]; then
+    echo "Usage:tiff_to_pyramid.bash <tmpdir> <full path to image source> <full path to output target> <save all working files - any value will do>";
```
Comment (fnibbit) [inline lib/nga.sh]: Usage message doesn't match script name.

File: lib/nga.sh [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+if [ ! -f $3 ]; then
+    echo "Could not create the output file.";
+    exit 1;
+fi
+
+rm -f $3
+
+tmpdir=$1
+# only process the first page of a potentially multi-page image document
```
Comment (fnibbit) [inline lib/nga.sh]: What does this comment apply to?

File: lib/nga.sh [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+# first, use vipsheader to read the bands
+CHANNELS=$(identify -format "%[channels]\n" ${input}[0] 2>/dev/null)
+echo "channels: ${CHANNELS}"
+if [ ${CHANNELS} = "srgba" ]; then
+    # we have to flatten the image to remove the alpha channel / trasparency before proceeding
+    echo "removing alpha channel from $input"
+    vips im_extract_bands $input ${input}.noalpha.tif 0 3   2>&1
+    if [ -z "${savefiles}" ]; then rm $input; fi
+    mv ${input}.noalpha.tif $input
```
Comment (fnibbit) [inline lib/nga.sh]: Looks like the original file gets overwritten whether or not ${savefiles} is set -- am I misreading this?  I see that ${savefiles} is unset when we use this tool from lib/yale_convert.rb though.

File: lib/nga.sh (hunk 3 above)
Comment (mlooney) [reply lib/nga.sh]: I have no idea. this is basically a copy of the NGA's script, with minor adjustments to paths & such

File: lib/nga.sh (hunk 4 above)
Comment (mlooney) [reply lib/nga.sh]: yeah, lots of fun stuff in this script. it'll also delete itself if temp is an empty string...

File: lib/nga.sh (hunk 1 above)
Comment (mark-dce) [reply lib/nga.sh]: It feels like this and the next comment can be resolved by renaming the script `tiff_to_pyramid.bash`.  Let's do that.

File: lib/nga.sh (hunk 2 above)
Comment (mark-dce) [reply lib/nga.sh]: It feels like this and the previous comment can be resolved by renaming the script `tiff_to_pyramid.bash`.

File: lib/nga.sh (hunk 3 above)
Comment (mark-dce) [reply lib/nga.sh]: Mike was very clear about wanting to run the NGA scripts unmodified.   Let's ensure that the rest of our code encapsulates them cleanly, but let's not attempt to modify their internals (even opaque comments) and redirect any questions about how the code functions back to whoever made the scripts at the NGA

File: lib/nga.sh (hunk 4 above)
Comment (mark-dce) [reply lib/nga.sh]: Let's make sure we cleanly encapsulate these scripts when calling them, but try not to modify them from the NGA originals.

**Review by mark-dce — CHANGES_REQUESTED** [review]
Let's rename the script to `tiff_to_pyramid.bash`.  Beyond that, this looks like a good place to start, so let's get it merged.

**Review by fnibbit — APPROVED** [review]

**Review by mark-dce — APPROVED** [review]

**Pull request merged**